### ISSUE TITLE
Error when using root_open: 'TDirectory' object has no attribute 'func'

### ISSUE THREAD

**Issue by beojan** [rootpy/rootpy#747]
As above:

`AttributeError: 'TDirectory' object has no attribute 'func'`

**Comment by ndawe** [comment]
Which version of ROOT and python?

**Comment by beojan** [comment]
ROOT 6.09/02, Python 3.6.1

**Comment by ndawe** [comment]
Can you paste the full stacktrace?

**Comment by ndawe** [comment]
(sorry, clicked wrong button and reopened issue)

**Comment by beojan** [comment]
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-ab852ddef913> in <module>()
----> 1 file = root_open('out.root')

/usr/lib/python3.6/site-packages/rootpy/io/file.py in root_open(filename, mode)
    135 
    136     filename = expand_path(filename)
--> 137     prev_dir = ROOT.gDirectory.func()
    138     root_file = ROOT.R.TFile.Open(filename, mode)
    139     if not root_file:

AttributeError: 'TDirectory' object has no attribute 'func'
```

**Comment by ndawe** [comment]
I haven't tested 6.09/02 yet. Looks like the ROOT devs must have removed that attr. This will be an easy fix.

**Comment by beojan** [comment]
The `.func` attribute is used elsewhere too (plotting for instance).

**Comment by ndawe** [comment]
True, but I can just patch one in on rootpy.ROOT and fix it everywhere.

**Comment by ndawe** [comment]
Fixed in #748

**Comment by ndawe** [comment]
I suggest avoiding 6.09/02 for now:

https://sft.its.cern.ch/jira/browse/ROOT-8789
